test/e2e/cli/cli_test.go

```diff
@@ -461,6 +463,88 @@ var _ = Describe("cli operation", func() {
+			Expect(err).NotTo(HaveOccurred())
+			Expect(out).To(MatchRegexp(resourceCreated))
+			fleetA := harness.GetFleetByYaml(uniqueFleetAYAML)
+			fleetAName := *fleetA.Metadata.Name
+			fleetACount := (*fleetA.Status.DevicesSummary).Total
+			fleetDevicesCount[fleetAName] = fleetACount
+
+			// Checking if number of devices is shown
+			By("Checking if number of devices is shown")
+			notMatched, err := CheckDevicesInFleetStatus(harness, fleetDevicesCount)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(notMatched).To(BeEmpty())
+
+			// Creating a device in the same fleet
+			By("Creating a device in the same fleet")
+			uniqueDeviceYAML, err := util.CreateUniqueYAMLFile("device.yaml", testID)
+			Expect(err).ToNot(HaveOccurred())
+			defer util.CleanupTempYAMLFile(uniqueDeviceYAML)
+
+			// Checking the device was created and updating in map
+			out, err = harness.ManageResource("apply", uniqueDeviceYAML)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(out).To(MatchRegexp(resourceCreated))
+			fleetACount = (*fleetA.Status.DevicesSummary).Total
+			fleetDevicesCount[fleetAName] = fleetACount
+
+			// Checking if the number of devices is shown and changed
+			By("Checking if number of devices is shown and changed")
+			notMatched, err = CheckDevicesInFleetStatus(harness, fleetDevicesCount)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(notMatched).To(BeEmpty())
+
+			// Creating another fleet
+			By("Creating another fleet")
+			uniqueFleetBYAML, err := util.CreateUniqueYAMLFile("fleet-b.yaml", testID)
+			Expect(err).ToNot(HaveOccurred())
+			defer util.CleanupTempYAMLFile(uniqueFleetBYAML)
+
+			// Checking the fleet was created and updating in map
+			out, err = harness.ManageResource("apply", uniqueFleetBYAML)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(out).To(MatchRegexp(resourceCreated))
+			fleetB := harness.GetFleetByYaml(uniqueFleetBYAML)
+			fleetBName := *fleetB.Metadata.Name
+			fleetBCount := (*fleetB.Status.DevicesSummary).Total
+			fleetDevicesCount[fleetBName] = fleetBCount
+
+			// Checking if the number of devices is shown for both fleets
+			By("Checking if number of devices is shown for both fleets")
+			notMatched, err = CheckDevicesInFleetStatus(harness, fleetDevicesCount)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(notMatched).To(BeEmpty())
+
+			// Deleting a device from the first fleet
+			By("Deleting a device from the first fleet")
+			out, err = harness.ManageResource("delete", uniqueFleetBYAML)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring("completed"))
+			fleetACount = (*fleetA.Status.DevicesSummary).Total
+			fleetDevicesCount[fleetAName] = fleetACount
+
+			// Check if the number is changed after deletion
+			By("Checking if number of devices is shown and changed")
+			notMatched, err = CheckDevicesInFleetStatus(harness, fleetDevicesCount)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(notMatched).To(BeEmpty())
```

Comment:
You may want to output the difference in case there is one, otherwise you can just use a counter